component/LinkableText/LinkableText.tsx

```diff
@@ -0,0 +1,25 @@
+import Link from "next/link";
+
+const LinkableText = ({text}: {text: string}) => {
+    const sanitizeNewLine = text?.split('\n')
+    const regexUrl = /^((?:https?:\/\/)?[^./]+(?:\.[^./]+)+(?:\/.*)?)$/
+
+    const textComponent = sanitizeNewLine?.map((textline, index) => {
+        const textLineSplit = textline?.split(' ')
+        return <p key={textline}>
+            {textLineSplit?.map((word, wordIndex) => {
+                if(wordIndex < textLineSplit?.length - 1) word += ' '
+                const hrefUrl = (word?.includes('http') || word?.includes('https')) ? word : `https://${word}`
+                if(regexUrl?.test(word)) return <Link className="underline text-signedPrimary" href={hrefUrl}>{word}</Link>
+                return word
+            })}
+        </p>
+        
+    })
+
+    return <>
+        {textComponent}
+    </>
+}
+
+export default LinkableText;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

It's good practice to end files with a newline character to avoid potential issues with certain tools that expect or require it. Please add a newline at the end of the file.
```diff
- export default LinkableText;
+ export default LinkableText;
+
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->